geoapi/tasks/external_data.py

```diff
@@ -118,6 +114,11 @@ def import_from_agave(user: User, systemId: str, path: str, proj: Project):
                     tmpFile.close()
                 else:
                     continue
+                # Save the row in the database that marks this file as already imported so it doesn't get added again
+                targetFile = ImportsService.createImportedFile(proj.id, systemId, str(item.path), item.lastModified)
```

Comment:
I wonder how we should handle when users delete a feature on an observable project.

I wonder if we should treat observable projects differently.  Maybe, we should display it differently in the frontend and disable some things for this map (feature deletion in the front end and in the back end, file adding in the front end and in the backend).  We would convey that the map is a one-to-one mapping of the directory and that users have to make direct changes (like adding or deleting files) to the DS directory.

Reply:
If a project is deleted, we should probably delete the ImportedFile rows that are related to that project.  Should we just add a ticket for now?

Reply:
> If a project is deleted, we should probably delete the ImportedFile rows that are related to that project. Should we just add a ticket for now?

The ondelete="CASCADE" on the project_id should handle it

Reply:
Yeah we should start thinking of how to do "published" projects that are read only. How much time would it take to implement what you mentioned though? I like the idea, I'm just lazy.

Reply:
Making a list to send you via slack to discuss